### PR TITLE
Use new markdown link checker

### LINKED ISSUE #602: Improve markdown link check
Maybe, our [markdown link check](https://github.com/gaurav-nelson/github-action-markdown-link-check) can be replaced by its successor: https://github.com/UmbrellaDocs/linkspector

### PR DESCRIPTION
**Pull request by Markus1812**
Updates our Github Action to check links in markdown with its successor https://github.com/UmbrellaDocs/linkspector using their new Github Action https://github.com/UmbrellaDocs/action-linkspector.

The new action adds reviews for broken links like this:
image

resolves #602

### CODE REVIEW

File: .github/workflows/markdown-link-check.yml [hunk 1]
```diff
@@ -5,14 +5,17 @@ on:
...
-    - uses: actions/checkout@main
-    - uses: gaurav-nelson/github-action-markdown-link-check@v1
-      with:
-        base-branch: 'main'
+      - uses: actions/checkout@v4
+      - name: Run linkspector
+        uses: umbrelladocs/action-linkspector@v1
+        with:
+          reporter: github-pr-review
+          fail_on_error: true
```
Comment (albertmink) [inline .github/workflows/markdown-link-check.yml:21]: It is convenient that files end with an empty line. So just add one.

Comment (Markus1812) [reply .github/workflows/markdown-link-check.yml]: Thanks, added an empty line. Just found out why: https://stackoverflow.com/a/729795

**Review by albertmink — APPROVED** [review]
LGTM

**Review by schneidermic0 — APPROVED** [review]
LGTM

Thanks, @Markus1812

**Pull request merged**